spec/system/custom_sidebar_sections_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+describe "Custom sidebar sections", type: :system, js: true do
+  fab!(:user) { Fabricate(:user) }
+  let(:section_modal) { PageObjects::Modals::SidebarSectionForm.new }
+  let(:sidebar) { PageObjects::Pages::CustomSectionsSidebar.new }
+
+  before do
+    ### TODO remove when enable_custom_sidebar_sections SiteSetting is removed
+    group = Fabricate(:group)
+    group_user = Fabricate(:group_user, group: group, user: user)
+    SiteSetting.enable_custom_sidebar_sections = group.id.to_s
+    sign_in user
+  end
+
+  it "allows the user to create custom section" do
+    visit("/latest?enable_sidebar=1")
+    sidebar.open_new_custom_section
+
+    expect(page).to have_css(".sidebar-section-form-modal")
```

Comment:
Since we have a `PageObjects::Modals::SidebarSectionForm`, I think we can extract the visibility check into a method in that object.

`expect(section_modal).to be_visible` or something like that?